### ISSUE TITLE
Skips up when the anchor links are offset from the top and user has already scrolled

### ISSUE THREAD

**Issue by keirstaple** [Nolanus/ng2-page-scroll#15]
Hey thanks so much for providing this directive! It works really well when the links to the anchors are at the very top of the page (startScrollTop === 0). However, for our site the links will be offset halfway down the page. When the user scrolls partway down the page (startScrollTop > 0) and clicks a link, the page will scroll to the anchor and then skip back up a bit. Any help would be great! Here is some code to give you and idea of what I'm talking about:

```
<p>1</p><p>1</p><p>1</p><p>1</p><p>1</p>
<a pageScroll [pageScrollDuration]="1/6*1000" href="#anchor1" class="goToAnchor">Go to Anchor 1</a>
<a pageScroll [pageScrollDuration]="2/6*1000" href="#anchor2" class="goToAnchor">Go to Anchor 2</a>
<a pageScroll [pageScrollDuration]="3/6*1000" href="#anchor3" class="goToAnchor">Go to Anchor 3</a>
<a pageScroll [pageScrollDuration]="4/6*1000" href="#anchor4" class="goToAnchor">Go to Anchor 4</a>
<a pageScroll [pageScrollDuration]="5/6*1000" href="#anchor5" class="goToAnchor">Go to Anchor 5</a>
<a pageScroll [pageScrollDuration]="6/6*1000" href="#anchor6" class="goToAnchor">Go to Anchor 6</a>

<div id="anchor1"><h1>Anchor 1</h1></div>
<div id="anchor2"><h1>Anchor 2</h1></div>
<div id="anchor3"><h1>Anchor 3</h1></div>
<div id="anchor4"><h1>Anchor 4</h1></div>
<div id="anchor5"><h1>Anchor 5</h1></div>
<div id="anchor6"><h1>Anchor 6</h1></div>

<p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p><p>1</p>
```

**Comment by keirstaple** [comment]
Awesome! Thanks for the follow-up. This fix worked for the most part. However, the issue now is that it scrolls past the anchor and then jumps back to the top of the anchor once the scroll is complete. Is there a way to prevent this and have it immediately stop on the anchor once it's at the top of the screen?

**Comment by keirstaple** [comment]
Let me know if I should open a new issue

**Comment by Nolanus** [comment]
Hi.
Sorry for the problem not being solved by the latest changes. Could you please create a plunkr/jsfiddle that reproduces the error? Don't forget to refer to the [git repo and commit hash in package.json](https://docs.npmjs.com/files/package.json#git-urls-as-dependencies) directly, as the latest changes are not yet published to npm (there still is another issue I want to fix first).
